customLabels.go

```diff
@@ -0,0 +1,48 @@
+package main
+
+import (
+	"encoding/json"
+	go_cache "github.com/patrickmn/go-cache"
+	"strings"
+	"time"
+	plog "github.com/prometheus/common/log"
+)
+
+// Custom labels for ConsumerGroups
+type CustomCGLagLabels struct {
+	labelByPrefix			map[string]string
+	labelCache			*go_cache.Cache
+}
+
+func NewCustomCGLagLabels(config string) (*CustomCGLagLabels, error){
+	labelByOwner := make(map[string][]string)
+	err := json.Unmarshal([]byte(config), &labelByOwner)
+	if err != nil {
+		return nil, err
+	}
+
+	labelByPrefix := make(map[string]string)
+	for owner, startWith := range  labelByOwner{
+		for _, startWith := range startWith {
+			labelByPrefix[startWith] = owner
+		}
+	}
+	labelCache := go_cache.New(1440*time.Minute, 60*time.Minute)
+	return &CustomCGLagLabels{labelByPrefix: labelByPrefix, labelCache: labelCache}, nil
+}
+
+func (c *CustomCGLagLabels) FetchLabel(groupId string) string {
+	var owner, found = c.labelCache.Get(groupId)
```

Comment:
I expire the cache because is based on consumergroup name, so if the broker runs for long time and the client creates new consumer groups we will keep growing and growing (adding more keys as the consumer groups in general are random names). The doc says if you don't add expiration then you have to manually clean it